File: app/src/main/java/org/androidforfun/droids/model/ShapeL.java
/*
 *  Copyright (C) 2016 Salvatore D'Angelo
 *  This file is part of Droids project.
 *
 *  Droids is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  Droids is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License.
 */
package org.androidforfun.droids.model;

/*
 * This is a L shape.
 */
public class ShapeL extends Shape {
    // An L-Shape has 4 rotation cycle, so it needs
    // to rotation to complete a cycle.
    public ShapeL() {
        super(2, 3);
        rotation_block = blocks[1];
        rotation_cycle = 4;
    }

    // The x,y position of each block is adjusted according to
    // the shape position and size. The adjust is first applied
    // assuming the shape is its first position. Then it is rotated.
    // The color is orange.
    public Block[] getBlocks() {
        blocks[0].setX(x);
        blocks[0].setY(y);
        blocks[1].setX(x);
        blocks[1].setY(blocks[0].getY() + 1);
        blocks[2].setX(x);
        blocks[2].setY(blocks[1].getY() + 1);
        blocks[3].setX(x + 1);
        blocks[3].setY(blocks[2].getY());

        apply_rotation();

        for (Block block : blocks) {
            block.setColor(0xffff7f00);
        }
        return blocks;
    }
}

File: app/src/main/kotlin/org/code4projects/droids/model/ShapeL.kt
/*
 * Copyright (c) 2016-2026 Salvatore D'Angelo
 * Licensed under the MIT License. See the LICENSE file in the project root.
 */
package org.code4projects.droids.model

/*
 * This is a L shape.
 */
class ShapeL : Shape(2, 3) {
    // An L-Shape has 4 rotation cycle, so it needs
    // one rotation to complete a cycle.
    init {
        rotationBlock = blockArray[1]
        rotationCycle = 4
    }

    // The x,y position of each block is adjusted according to
    // the shape position and size. The adjust is first applied
    // assuming the shape is its first position. Then it is rotated.
    // The color is orange.
    override fun getBlocks(): Array<Block> {
        blockArray[0].x = x
        blockArray[0].y = y
        blockArray[1].x = x
        blockArray[1].y = blockArray[0].y + 1
        blockArray[2].x = x
        blockArray[2].y = blockArray[1].y + 1
        blockArray[3].x = x + 1
        blockArray[3].y = blockArray[2].y

        applyRotation()

        for (block in blockArray) {
            block.color = 0xffff7f00L.toInt()
        }
        return blockArray
    }
}
